lib/web3.js

```diff
@@ -78,56 +79,60 @@ var setupProperties = function (obj, properties) {
 };
 
 /// setups web3 object, and it's in-browser executed methods
-var web3 = {};
-web3.providers = {};
-web3.currentProvider = null;
-web3.version = {};
-web3.version.api = version.version;
-web3.eth = {};
+function Web3(provider) {
```

Comment:
So you need to instantiate `Web3`?
So you would like to have multiple instances, with different providers running in the same app?

web3 is designed to be an object like the `chrome` or later `mist` object. It gives "native" functions and the dapp shouldn't even care how it connects.